### PR TITLE
Model updates - added node_modules to .gitignore

### PR DESCRIPTION
**Pull request by hstevemcdonald**

### CODE REVIEW

File: ListOfArtists.html [hunk 1]
```diff
@@ -11,7 +11,7 @@
 <script src="https://cdnjs.cloudflare.com/ajax/libs/materialize/0.98.2/js/materialize.min.js"></script>
```
Comment (lizzkats) [inline ListOfArtists.html:12]: I think this file needs to be in EJS if it's the 'artists' page

File: app/model/album.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  this.albums.map(function(val) {
+   if (val.artist_id == id)
+   {
+      count++;
+   }
+  })
+  return count;
+}
+
+Album.prototype.loadAll = function(callback) {
```
Comment (lizzkats) [inline app/model/album.js]: Consider removing this parameter as it's not being used.

File: app/model/album.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+var app = express();
+var fs =require('fs');
+
+Album = function() {
+  this.albums = [];
+  this.loadAll();
+}
+
+Album.prototype.get =function(id) {
+  if (!id) {return false;}
```
Comment (lizzkats) [inline app/model/album.js]: Consider adding whitespace to increase readability for the purpose of maintainability.

File: app/model/album.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  return this.albums[id];
+}
+
+Album.prototype.countAlbums = function(id)
+{
```
Comment (lizzkats) [inline app/model/album.js]: This is inconsistent styling so you should either fix this one or change the other opening curly braces.

File: app/model/album.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  console.log(this.albums)
```
Comment (lizzkats) [inline app/model/album.js]: You should consider not committing your console logs for production code.

File: app/model/album.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  var count = 0;
+  this.albums.map(function(val) {
+   if (val.artist_id == id)
+   {
+      count++;
+   }
+  })
+  return count;
```
Comment (lizzkats) [inline app/model/album.js]: Whenever I see the pattern:
```js
// variable declaration
// iteration
// return variable
```
I replace it with the more concise  `reduce`:
```js
return this.albums.reduce( function( memo, album ) {
  return memo + ( album.artist_id === id ? 1 : 0 )
}, 0 );
```

File: app/model/album.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+Album.prototype.loadAll = function(callback) {
+  var _this = this;
+  console.log("Reading album files...")
```
Comment (lizzkats) [inline app/model/album.js]: consider removing console.log

File: app/model/artist.js [hunk 8]
```diff
@@ -2,28 +2,41 @@ var express = require('express');
...
+    this.artists = data;
+  }
+}
+
 Artist.prototype.loadAll = function(callback)
 {
-  console.log("Reading files...")
+  var _this = this;
+
+  console.log("Reading artist files...")
```
Comment (lizzkats) [inline app/model/artist.js]: consider removing console.log

File: app/model/song.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+ this.songs = [];
+ this.loadAll();
+}
+
+Song.prototype.get =function(id) {
+  if (!id) {return false;}
+  return this.songs[id];
+}
+
+Song.prototype.loadAll = function(callback) {
```
Comment (lizzkats) [inline app/model/song.js]: This behavior looks repeated from the artists.js and albums.js, consider refactoring to some explicit module and importing it into artist, song and album.

File: .gitignore [hunk 10]
```diff
@@ -49,3 +49,5 @@ fabric.properties
 aback-kingfisher-music-app.iml
```
Comment (lizzkats) [inline .gitignore]: You can use *.iml instead of specifying file name here :)

**Review by lizzkats — COMMENTED** [review]
Good job! Make sure you remove console.logs before committing production code. Also consistent spacing and curly braces. 👍

File: .gitignore (hunk 10 above)
Comment (latoyazamill) [reply .gitignore]: Complete :)

File: app/model/artist.js (hunk 8 above)
Comment (latoyazamill) [reply app/model/artist.js]: Complete :)

File: ListOfArtists.html (hunk 1 above)
Comment (latoyazamill) [reply ListOfArtists.html:12]: Complete :)

**Pull request merged**